tasks/verify_provision.yml

```diff
@@ -0,0 +1,61 @@
+---
+
+- name: "Verify {{ app_name }} objects exist"
+  command: "oc get {{ item }} -n {{ namespace }} {{ app_name }}"
+  register: object
+  retries: 10
+  delay: 5
+  until: object.rc == 0
+  failed_when: object.rc != 0
+  with_items:
+    - DeploymentConfig
+    - Service
+    - Route
+  when: cluster == "openshift"
+
+- name: "Verify {{ app_name }} objects exist"
+  command: "kubectl get {{ item }} -n {{ namespace }} {{ app_name }}"
+  register: object
+  retries: 10
+  delay: 5
+  until: object.rc == 0
+  failed_when: object.rc != 0
+  with_items:
+    - Deployment
+    - Service
+  when: cluster == "kubernetes"
+
+- name: "Wait for deployment config to be available"
+  command: >-
```

Comment:
I think chouseknecht has added a lookup module for this, probably a good place to showcase it: https://docs.ansible.com/ansible/devel/plugins/lookup/openshift.html

I think ansible also has a jsonquery/path filter built in to it, probably should add the dependency to apb-base since it's pretty useful.

Reply:
@jmontleon 
I did a cursory look to see if jmespath was available in the EPEL repo for centos in order to update the apb-base image and it wasn't there. I see a few bugs related to this https://bugzilla.redhat.com/show_bug.cgi?id=1484910 but I don't see how we could update the apb-base with jmespath. If we can't easily add it then I think we should push off the lookup stuff.